packages/context/src/ui5-yaml.ts

```diff
@@ -30,35 +21,35 @@ export async function initializeUI5YamlData(
   const readUI5YamlPromises = map(ui5YamlDocuments, async (ui5YamlDoc) => {
     const response = await readUI5YamlFile(ui5YamlDoc);
 
-    // Parsing of ui5.yaml failed because the file is invalid
-    if (response !== "INVALID") {
-      ui5YamlData[ui5YamlDoc] = response;
+    if (response) {
+      cache.setYamlDetails(ui5YamlDoc, response);
+      console.info("ui5.yaml data initialized", { ui5YamlDoc });
     }
   });
 
-  getLogger().info("ui5.yaml data initialized", { ui5YamlDocuments });
+  console.info("list of ui5.yaml files", { ui5YamlDocuments });
   return Promise.all(readUI5YamlPromises);
 }
 
 export function getUI5FrameworkForXMLFile(xmlPath: string): UI5Framework {
   const ui5YamlFilesForCurrentFolder = filter(
-    Object.keys(ui5YamlData),
+    cache.getYamlDetailsEntries(),
     (ui5YamlPath) => xmlPath.startsWith(dirname(ui5YamlPath))
   );
 
   const closestUI5YamlPath = maxBy(
     ui5YamlFilesForCurrentFolder,
     (ui5YamlPath) => ui5YamlPath.length
   );
-
-  return closestUI5YamlPath
-    ? ui5YamlData[closestUI5YamlPath].framework
-    : "SAPUI5";
+  if (closestUI5YamlPath) {
+    return cache.getYamlDetails(closestUI5YamlPath)?.framework ?? "OpenUI5";
+  }
+  return "OpenUI5";
```

Comment:
Why did we change the default framework here?

Reply:
Thanks. We should not. :) This method is not used, so removed it completely. It is handled in `getYamlDetails` methods and default framework is `SAPUI5`